src/ui/common/src/components/integrations/dialogs/athenaDialog.tsx

```diff
@@ -25,49 +33,62 @@ const Placeholders: AthenaConfig = {
   output_location: 's3://bucket/path/to/folder/',
 };
 
-type Props = {
-  onUpdateField: (field: keyof AthenaConfig, value: string) => void;
-  value?: AthenaConfig;
-  editMode: boolean;
-};
+// type Props = {
+//   onUpdateField: (field: keyof AthenaConfig, value: string) => void;
+//   value?: AthenaConfig;
+//   editMode: boolean;
+// };
 
-export const AthenaDialog: React.FC<Props> = ({
-  onUpdateField,
-  value,
-  editMode,
+export const AthenaDialog: React.FC<IntegrationDialogProps> = ({
+  editMode = false,
 }) => {
+  const { register, getValues, setValue } = useFormContext();
+
+  // Need state variable to change tabs, as the formContext doesn't change as readily.
+  const [currentTab, setCurrentTab] = useState(AWSCredentialType.AccessKey);
+
   const [fileName, setFileName] = useState<string>(null);
 
   const setFile = (fileData: FileData | null) => {
     setFileName(fileData?.name ?? '');
-    onUpdateField('config_file_content', fileData?.data);
+    console.log('fileData', fileData);
+
+    //onUpdateField('config_file_content', fileData?.data);
+    setValue('config_file_content', fileData?.data);
   };
 
   const fileData =
-    fileName && !!value?.config_file_content
+    fileName && !!getValues('config_file_content')
       ? {
           name: fileName,
-          data: value.config_file_content,
+          data: getValues('config_file_content'),
```

Comment:
Debugging notes on fixing the file upload errors:
TODO: log config_file_conttent. see what we're getting back here after upload.